roles/custom/matrix-dendrite/templates/dendrite/dendrite.yaml.j2

```diff
@@ -414,3 +414,16 @@ tracing:
 # Logging configuration, in addition to the standard logging that is sent to
 # stdout by Dendrite.
 logging: []
+
+# Users who register on this homeserver will automatically be joined to the rooms listed under "auto_join_rooms" option.
+# By default, any room aliases included in this list will be created as a publicly joinable room
+# when the first user registers for the homeserver. If the room already exists,
+# make certain it is a publicly joinable room, i.e. the join rule of the room must be set to 'public'.
+# As Spaces are just rooms under the hood, Space aliases may also be used.
+#auto_join_rooms:
+#  - "#main:matrix.org"
+
+{% if matrix_dendrite_auto_join_rooms|length > 0 %}
+auto_join_rooms:
+{{ matrix_dendrite_auto_join_rooms|to_nice_yaml(indent=2, width=999999) }}
+{% endif %}
```

Comment:
This is in the wrong place.

If you look at the example configuration, you'll see that `auto_join_rooms` is **not** top-level, but is nested under `userapi`.

https://github.com/matrix-org/dendrite/blob/efa50253f68f8ec1c6a9f4a8d82b329512027b00/dendrite-sample.monolith.yaml#L303-L325

Because of this nesting and our rules for naming variables, the variable name should also be called `matrix_dendrite_userapi_auto_join_rooms`.

You also don't need an `if` check for `length`, I suppose.

You could probably just do:

```yaml
userapi:
  # Lots of stuff here..
  auto_join_rooms: {{ matrix_dendrite_userapi_auto_join_rooms | to_json }}
```

Reply:
Thank you so much! 
Just double checking, did you mean under `user_api`?

Reply:
Yes, you're right - it should be `user_api`